contracts/token/ERC721/ERC721BasicToken.sol

```diff
@@ -0,0 +1,279 @@
+    address owner = ownerOf(_tokenId);
+    require(_to != owner);
+    require(msg.sender == owner || isApprovedForAll(owner, msg.sender));
+
+    if (getApproved(_tokenId) != 0 || _to != 0) {
+      tokenApprovals[_tokenId] = _to;
+      Approval(owner, _to, _tokenId);
+    }
+  }
+
+  /**
+   * @dev Gets the approved address for a token ID, or zero if no address set
+   * @param _tokenId uint256 ID of the token to query the approval of
+   * @return address currently approved for a the given token ID
+   */
+  function getApproved(uint256 _tokenId) public view returns (address) {
+    return tokenApprovals[_tokenId];
+  }
+
+
+  /**
+  * @dev Sets or unsets the approval of a given operator
+  * @dev An operator is allowed to transfer all tokens of the sender on their behalf
+  * @param _to operator address to set the approval
+  * @param _approved representing the status of the approval to be set
+  */
+  function setApprovalForAll(address _to, bool _approved) public {
+    require(_to != msg.sender);
+    operatorApprovals[msg.sender][_to] = _approved;
+    ApprovalForAll(msg.sender, _to, _approved);
+  }
+
+  /**
+   * @dev Tells whether an operator is approved by a given owner
+   * @param _owner owner address which you want to query the approval of
+   * @param _operator operator address which you want to query the approval of
+   * @return bool whether the given operator is approved by the given owner
+   */
+  function isApprovedForAll(address _owner, address _operator) public view returns (bool) {
+    return operatorApprovals[_owner][_operator];
+  }
+
+  /**
+  * @dev Transfers the ownership of a given token ID to another address
+  * @dev Usage of this method is discouraged, use `safeTransferFrom` whenever possible
+  * @dev Requires the msg sender to be the owner, approved, or operator
+  * @param _from current owner of the token
+  * @param _to address to receive the ownership of the given token ID
+  * @param _tokenId uint256 ID of the token to be transferred
+  */
+  function transferFrom(address _from, address _to, uint256 _tokenId) public canTransfer(_tokenId) {
+    clearApprovalAndTransfer(_from, _to, _tokenId, "", false);
+  }
+
+  /**
+  * @dev Safely transfers the ownership of a given token ID to another address
+  * @dev If the target address is a contract, it must implement `onERC721Received`,
+  *  which is called upon a safe transfer, and return the magic value
+  *  `bytes4(keccak256("onERC721Received(address,uint256,bytes)"))`; otherwise,
+  *  the transfer is reverted.
+  * @dev Requires the msg sender to be the owner, approved, or operator
+  * @param _from current owner of the token
+  * @param _to address to receive the ownership of the given token ID
+  * @param _tokenId uint256 ID of the token to be transferred
+  */
+  function safeTransferFrom(address _from, address _to, uint256 _tokenId) public canTransfer(_tokenId) {
+    clearApprovalAndTransfer(_from, _to, _tokenId, "", true);
```

Comment:
should this be `safeTransferFrom(_from, _to, _tokenId, "");` to make it clear that its implementation is the same as the other function, but with empty data? Is there a gas concern with this approach?